src/System.Private.ServiceModel/tools/PRService/pr.ashx

```diff
@@ -60,26 +62,35 @@ public class PullRequestHandler : IHttpHandler
             context.Response.Write(HttpUtility.HtmlEncode(result));
             return;
         }
-        else
+
+        success = false;
+
+        // Check inputs to the script. PR has precedence
+        uint pr;
+        if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
         {
-            // Check inputs to the script
-            uint pr;
-            if (!string.IsNullOrEmpty(prString) && uint.TryParse(prString, out pr))
-            {
-                success = SyncToPr(pr, result);
-            }
-            else
+            success = SyncToPr(pr, result);
+        }
+        else if (!string.IsNullOrEmpty(branchString))
```

Comment:
I agree that I _usually_ would validate right at the beginning, yes (see all the code I write for WCF itself :smile:). This time I feel like verifying prerequisites trumps.